tracing-subscriber/Cargo.toml

```diff
@@ -24,7 +24,7 @@ keywords = ["logging", "tracing", "metrics", "subscriber"]
 [features]
 
 default = ["env-filter", "smallvec", "fmt", "ansi", "chrono", "tracing-log", "json"]
-env-filter = ["matchers", "regex", "lazy_static"]
+env-filter = ["matchers", "regex", "lazy_static", "tracing"]
```

Comment:
Not sure `tracing` needs to be another feature flag on this. I don't see this flag being used elsewhere _and_ I think that a tracing dependency is a forgone conclusion for `tracing-subscriber`.

Reply:
This won't compile without adding a `tracing` dependency:

```
error[E0433]: failed to resolve: use of undeclared type or module `tracing`
   --> tracing-subscriber/src/filter/env/mod.rs:245:13
    |
245 |         use tracing::level_filters::STATIC_MAX_LEVEL;
    |             ^^^^^^^ use of undeclared type or module `tracing`
error[E0433]: failed to resolve: use of undeclared type or module `tracing`
   --> tracing-subscriber/src/filter/env/mod.rs:245:13
    |
245 |         use tracing::level_filters::STATIC_MAX_LEVEL;
    |             ^^^^^^^ use of undeclared type or module `tracing`
```

Reply:
I'd like to hear what @hawkw says, but I'm not sure `tracing` merits being an optional dependency.

Reply:
Oh I see, you want to make it always a dependency instead of optional. Sure, I can do that.

Reply:
I think the `tracing` dependency should be optional and only enabled by the "env-filter" feature. Otherwise, there's no reason to depend on `tracing` --- the rest of the crate only uses `tracing-core`.

I think @davidbarsky misinterpreted this change as adding a new _feature flag_ "tracing" that users could independently enable. That's not what this change is doing --- it simply makes the "env-filter" feature enable a dependency on `tracing` that isn't otherwise necessary.

IMO the original code was correct.

Reply:
Ok, I reverted it to the original code.